### PR TITLE
Skip Account Validation based on request parameters.

### PR DESCRIPTION
**Pull request by swasti29**
## Proposed changes

Skip Account Validation based on request parameters.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m [hunk 1]
```diff
@@ -186,7 +186,9 @@ - (NSDictionary *)defaultPayloadContents:(NSError **)error
         queryDictionary[MSID_NESTED_AUTH_BROKER_CLIENT_ID] = self.requestParameters.nestedAuthBrokerClientId;
         queryDictionary[MSID_NESTED_AUTH_BROKER_REDIRECT_URI] = self.requestParameters.nestedAuthBrokerRedirectUri;
     }
-
+    
+    [queryDictionary msidSetNonEmptyString:self.requestParameters.clientSku forKey:MSID_CLIENT_SKU_KEY];
```
Comment (Veena11) [inline IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m:190]: where/how client sku used for skipping account validation ?

**Review by Veena11 — APPROVED** [review]

File: IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m (hunk 1 above)
Comment (swasti29) [reply IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m:190]: Setting client sku here: https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1993/files#diff-558fd2bfadacab5121f3004397dd79c8e32c3e6c83e0d6cdc265ef814f6f1c9bR1209

This is also being used by MSAL CPP.

Comment (swasti29) [reply IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m:190]: This parameter has been added for future purposes if we need to do some specific task based on calling SDK.

Comment (Veena11) [reply IdentityCore/src/requests/broker/MSIDBrokerTokenRequest.m:190]: aah ok, thanks

**Comment by kaisong1990** [comment]
Looks good, please work with @ashok672 to do a e2e check once this is merged.

**Review by kaisong1990 — APPROVED** [review]

**Pull request merged**